### DISCUSSION TITLE
Add loading state while using useLoaderData and  loader object

### DISCUSSION THREAD

**Discussion by forinda** [remix-run/react-router#9985] [Proposals]
The element shown on the page is just a blank canvas when still fetching data in the loader.
I wanted a way to get the state of this when still fetching. I tried the `useNavigation ` but just getting `idle` everytime

**Comment by prvnbist** [comment]
I'm also stuck at this point, anyway to show loading state untill the `loader` finishes? @mjackson

**Comment by brophdawg11** [comment]
Can you provide a reproduction of your issue?  It's hard to debug further without seeing how you're application is setup.
